logictest: skip TestParallel under stressrace

asubiotto opened the pull request:
Normal logic tests are skipped under stressrace, and this commit does the
same for TestParallel.

Release note: None

Closes #56041

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56110)
<!-- Reviewable:end -->

Commit:
logictest: skip TestParallel under stressrace

Normal logic tests are skipped under stressrace, and this commit does the
same for TestParallel.

Release note: None

asubiotto commented:
TFTR

bors r=RaduBerinde
